Classes/Notifications/NotificationsViewController.swift

```diff
@@ -56,6 +56,28 @@ FlatCacheListener {
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if NotificationClient.readOnOpen(){
+            self.readOnOpen()
+        }
+        
+    }
+    
+   private func readOnOpen(){
```

Comment:
I have a hard time parsing what's happening in this function. Can we simplify this?

Reply:
Some header docs explaining what's going on might help too.